test/Tests/src/Data/Text/Encoding_Spec.enso

```diff
@@ -0,0 +1,84 @@
+from Standard.Base import all
+
+from Standard.Base.Data.Text.Encoding as Encoding_Module import Encoding, Encoding_Error
+import Standard.Base.Data.Text.Encoding as Encoding_Module
+
+import Standard.Test
+import Standard.Test.Problems
+
+spec =
+    Test.group "Encoding object" <|
+        Test.specify "Can get standard UTF encodings" <|
+            Encoding.utf_8 . should_equal (Encoding "UTF-8")
+            Encoding.utf_16_le . should_equal (Encoding "UTF-16LE")
+            Encoding.utf_16_be . should_equal (Encoding "UTF-16BE")
+            Encoding.utf_32_le . should_equal (Encoding "UTF-32LE")
+            Encoding.utf_32_be . should_equal (Encoding "UTF-32BE")
+
+        Test.specify "Catches invalid character sets" <|
+            invalid = Encoding "NotAValidCharacterSet"
+            invalid.to_java_charset . should_fail_with Illegal_Argument_Error
+
+        Test.specify "Can get full set of character sets" <|
+            character_sets = Encoding_Module.all_character_sets
+            character_sets.length . should_not_equal 0
+            character_sets.contains "UTF-8" . should_equal True
+
+        Test.specify "Can get full set of encoding objects" <|
+            encodings = Encoding_Module.all_encodings
+            encodings.length . should_not_equal 0
+            encodings.contains Encoding.utf_8 . should_equal True
+
+    Test.group "ASCII" <|
+        test = 'Hello World!'
+        test_ascii = [72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33]
+
+        invalid = 'Hello World!\uFFFD'
+        invalid_ascii = [72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33, -56]
+        invalid_ascii_out = [72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33, 63]
+
+        Test.specify "should dump ASCII bytes to a vector via encoding" <|
+            test.bytes Encoding.ascii . should_equal test_ascii
+
+        Test.specify "should convert an array of bytes to text" <|
+            Text.from_bytes test_ascii Encoding.ascii . should_equal test
+
+        Test.specify "Invalid ASCII should raise a warning when decoding" <|
+            action = Text.from_bytes invalid_ascii Encoding.ascii on_problems=_
+            tester result = result . should_equal invalid
+            problems = [Encoding_Error "Encoding issues at 12"]
+            Problems.test_problem_handling action problems tester
+
+        Test.specify "Invalid ASCII should raise a warning when encoding" <|
+            action = invalid.bytes Encoding.ascii on_problems=_
+            tester result = result . should_equal invalid_ascii_out
+            problems = [Encoding_Error "Encoding issues at 12"]
+            Problems.test_problem_handling action problems tester
+
+    Test.group "UTF_8" <|
```

Comment:
Can we get a test for UTF-16 and Windows-1251?

Also is it possible to create invalid UTF-8 codepoints so that the issue reporting for UTF-8 could be tested?